### PR TITLE
(BOLT-916) Remove ability accept file content from boltserver

### PR DESCRIPTION
**Pull request by adreyer**
Remove support for sending a "file" with the task in boltserver now that
we support downloading and caching files.

### CODE REVIEW

File: lib/bolt/task/puppet_server.rb [hunk 1]
```diff
@@ -11,19 +11,17 @@ def initialize(task_data, file_cache)
...
+      # puppetserver file entries have 'filename' rather then 'name'
       def update_file_data(task_data)
-        if task_data['files']
-          task_data['files'] = task_data['files'].map do |f|
-            { 'name' => f['filename'],
-              'path' => @file_cache.update_file(f) }
-          end
-        end
-
+        task_data['files'] = task_data['files'].map { |f| f['name'] = f['filename'] }
```
Comment (MikaelSmith) [inline lib/bolt/task/puppet_server.rb]: Could use `map!`.

File: lib/bolt/task/puppet_server.rb [hunk 2]
```diff
@@ -11,19 +11,17 @@ def initialize(task_data, file_cache)
         super(task_data)
       end
 
-      # TODO: currently this both converts filename -> name and fetches files to generate the path
-      # In the future we should not generate paths until after the implementation is selected to avoid
-      # unecessary fetching of files
+      # puppetserver file entries have 'filename' rather then 'name'
```
Comment (MikaelSmith) [inline lib/bolt/task/puppet_server.rb]: That's inconvenient.

File: lib/bolt/task/puppet_server.rb [hunk 3]
```diff
@@ -11,19 +11,17 @@ def initialize(task_data, file_cache)
...
-          end
-        end
-
+        task_data['files'] = task_data['files'].map { |f| f['name'] = f['filename'] }
         task_data
       end
+
+      # Compute local path and download files from puppetserver as needed
+      def file_path(file_name)
+        file = file_map[file_name]
```
Comment (MikaelSmith) [inline lib/bolt/task/puppet_server.rb]: Do we need to look up the file if `file['path']` is already set? Could use a `begin` block to wrap everything.

File: lib/bolt/transport/ssh.rb [hunk 4]
```diff
@@ -122,19 +122,11 @@ def run_script(target, script, arguments, options = {})
...
-          executable = implementation['path']
-          input_method = implementation['input_method']
-          extra_files = implementation['files']
-          write_executable = proc { |conn, dir| conn.write_remote_executable(dir, executable) }
-        end
+        implementation = task.select_implementation(target, PROVIDED_FEATURES)
+        executable = implementation['path']
+        input_method = implementation['input_method']
+        extra_files = implementation['files']
+        write_executable = proc { |conn, dir| conn.write_remote_executable(dir, executable) }
```
Comment (MikaelSmith) [inline lib/bolt/transport/ssh.rb]: The proc isn't really needed anymore, could move the call back to where it's used.

File: lib/bolt/transport/winrm.rb [hunk 5]
```diff
@@ -108,19 +108,11 @@ def run_script(target, script, arguments, _options = {})
...
-          executable = implementation['path']
-          input_method = implementation['input_method']
-          extra_files = implementation['files']
-          write_executable = proc { |conn, dir| conn.write_remote_executable(dir, executable) }
-        end
+        implementation = task.select_implementation(target, PROVIDED_FEATURES)
+        executable = implementation['path']
+        input_method = implementation['input_method']
+        extra_files = implementation['files']
+        write_executable = proc { |conn, dir| conn.write_remote_executable(dir, executable) }
```
Comment (MikaelSmith) [inline lib/bolt/transport/winrm.rb]: Don't need the proc anymore.

File: lib/bolt/task/puppet_server.rb (hunk 2 above)
Comment (adreyer) [reply lib/bolt/task/puppet_server.rb]: Yeah I discovered that a few months ago when working on cross platform tasks in puppetserver. It's a public API so we should probably change puppet.

File: lib/bolt/task/puppet_server.rb (hunk 1 above)
Comment (adreyer) [reply lib/bolt/task/puppet_server.rb]: yeah I'm mutating them anyway

File: lib/bolt/task/puppet_server.rb (hunk 3 above)
Comment (adreyer) [reply lib/bolt/task/puppet_server.rb]: We don't call update_file on the next line if path is set. I'm not sure what you're suggesting, this cuts out a line of code:
```
file_map[file_name]['path'] ||= @file_cache.update_file(file)
```

File: lib/bolt/task/puppet_server.rb (hunk 1 above)
Comment (adreyer) [reply lib/bolt/task/puppet_server.rb]: I think each is actually what I want.

File: lib/bolt_server/file_cache.rb [hunk 6]
```diff
@@ -19,38 +27,56 @@ def tmppath
...
-      uri = "#{@config.file_server_uri}#{path}"
+    def ssl_cert
+      @ssl_cert ||= File.read(@config.ssl_cert)
+    end
 
+    def ssl_key
+      @ssl_key ||= File.read(@config.ssl_key)
+    end
+
+    def client
```
Comment (adreyer) [inline lib/bolt_server/file_cache.rb]: I 'm not sure what http library we'd want to use but I think this should keep the connection open for 2s by default.

**Comment by MikaelSmith** [comment]
Some tests for pe-bolt-server in Orchestrator need to be fixed before this is merged.

**Comment by MikaelSmith** [comment]
Fix for winrm up at https://github.com/puppetlabs/bolt/pull/720.

**Comment by MikaelSmith** [comment]
I'm working on the Orchestrator test fix. The winrm fix is in.

**Comment by MikaelSmith** [comment]
We should merge https://github.com/puppetlabs/orchestrator/pull/930 before this.

File: spec/lib/bolt_spec/file_cache.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+      def setup
+        self
+      end
+
+      def update_file(file_data)
+        parts = file_data['filename'].split('/')
+        modulename = 'sample'
+        if parts.size == 1
+          File.join(@moduledir, modulename, 'tasks', parts[0])
```
Comment (MikaelSmith) [inline spec/lib/bolt_spec/file_cache.rb]: Not sure this will work on windows because it doesn't include an extension.

Comment (MikaelSmith) [reply spec/lib/bolt_spec/file_cache.rb]: Nevermind, misread what it does.

**Pull request merged**